README.md

```diff
@@ -113,58 +118,80 @@ const queue = new SQSMessageHandler({
+const sqsHandler = new SQSMessageHandler(config)
   .onMessage(async (ctx, message) => {
     // `ctx` contains the nice result of `createRunContext`:
     await ctx.doSomething();
+    // `ctx` contains a logger by default, which already includes niceties like
+    // the AWS request id
+    ctx.logger.info('processed SQS message');
+  })
+  // Add multiple message handlers for code organization.
+  .onMessage(async (ctx, message) => {
+    // do something else
+  });
 
+// Or with SNS - identical API
+const snsHandler = new SNSMessageHandler(config)
+  .onMessage(async (ctx, message) => {
+    // `ctx` contains the nice result of `createRunContext`:
+    await ctx.doSomething();
     // `ctx` contains a logger by default, which already includes niceties like
     // the AWS request id
-    ctx.logger.info('blah blah');
+    ctx.logger.info('processed SNS message');
   })
   // Add multiple message handlers for code organization.
   .onMessage(async (ctx, message) => {
     // do something else
   });
 
-// Provides a dead-simple API for creating the Lambda.
-export const handler = stream.lambda();
+// Both provide the same lambda() method
+export const sqsLambda = sqsHandler.lambda();
+export const snsLambda = snsHandler.lambda();
 ```
 
-`SQSMessageHandler` also comes with a nice helper for testing: `harness(...)`
+#### Shared Testing Harness
+
+Both handlers provide identical testing harnesses:
 
 ```typescript
-const context = {
-  doSomething: jest.fn()
-}
+const context = { doSomething: jest.fn() };
 
-const harness = queue.harness({
-  stringifyMessage: (message) => {
-    /* stringify from your custom type -> string */
-    return JSON.stringify(message)
-  },
-  /* optionally override the logger */
+// Same harness API for both handlers
+const testConfig = {
+  stringifyMessage: (message) => JSON.stringify(message),
   logger,
-  createRunContext: () => {
-    /* optionally override the context, to mock e.g. data sources */
-    return context;
-  }
-})
+  createRunContext: () => context,
+};
 
-test('something', async () => {
-  // Provides a simple `sendEvent` function
-  await harness.sendEvent({
-    messages: [
-      { /* message 1 */}
-      { /* message 2 */}
-      { /* message 3 */}
-    ]
-  })
+const sqsHarness = sqsHandler.harness(testConfig);
+const snsHarness = snsHandler.harness(testConfig);
 
-  expect(context.doSomething).toHaveBeenCalledTimes(3)
-})
+test('SQS messages', async () => {
+  await sqsHarness.sendEvent({
+    messages: [{ id: 1 }, { id: 2 }],
+  });
+  expect(context.doSomething).toHaveBeenCalledTimes(2);
+});
+
+test('SNS messages', async () => {
+  await snsHarness.sendEvent({
+    messages: [{ id: 1 }, { id: 2 }],
+  });
+  expect(context.doSomething).toHaveBeenCalledTimes(2);
+});
 ```
 
+#### Key Differences
+
+- **`SQSMessageHandler`**: Processes SQS queue messages, uses `MessageGroupId` for ordering
+- **`SNSMessageHandler`**: Processes SNS topic notifications, uses `MessageId` for ordering
```

Comment:
What I didn't follow here is that for the use case in this epic we'd still use the `SQSMessageHandler` to consume the published SNS event. So, where would we be using the `SNSMessageHandler`?

Reply:
You're right, I thought we needed to process SNS messages into SQS records from a subscription, not that the receiver would be SQS. 

I'll see myself out🚶‍➡️